perf-test-jenkins/Main.groovy

```diff
@@ -169,23 +170,28 @@ private String runTests(File windup, String name, File inputFile) {
         detailedStatsFile.eachLine { line, number ->
             if (number == 1)
                 return
-            //Type~Number Of Executions~Total Milliseconds~Milliseconds per execution
-            def match = line =~ /(.*?)~(.*?)~(.*?)~(.*)/;
-            if (match.matches()) {
-                def detailedStatName = match.group(1);
-                def numberOfExecs = Integer.valueOf(match.group(2));
-                def totalMillis = Integer.valueOf(match.group(3));
-                //println("Name: " + detailedStatName + " # " + numberOfExecs + " totalMillis: " + totalMillis);
-                def detailedStatRow = detailedStats[detailedStatName];
-                if (!detailedStatRow) {
-                    detailedStatRow = [:];
-                    detailedStatRow["executions"] = numberOfExecs;
-                    detailedStatRow["total"] = totalMillis;
-                    detailedStats[detailedStatName] = detailedStatRow;
-                } else {
-                    detailedStatRow["executions"] = detailedStatRow["executions"] + numberOfExecs;
-                    detailedStatRow["total"] = detailedStatRow["total"] + totalMillis;
-                }
+            // Number Of Executions, Total Milliseconds, Milliseconds per execution, "Type"
+            def match = line =~ /([^,]*?),\s*([^,]*?),\s*([^,]*?),\s*(?:([^",]+)|(?:"([^"]+(?:\\")?)+"))$/; //"
```

Comment:
The first 3 are numbers, so they really should not.
The 4th is either a string, or can have commas but then it will be escaped.
See http://fiddle.re/2mm08a